### DISCUSSION TITLE
Access current route

### DISCUSSION THREAD

**Discussion by aldo-roman** [withastro/roadmap#970] [Proposal]
### Body

# Summary

To have a property `Astro.route` that contains the current route being rendered.

# Background & Motivation

When rendering a navbar statically, I'd like to know which route is currently active. Right now, I can only know which URL is active via `Astro.url`. Unfortunately, this does not allow to know which route is being rendered.

# Goals

- To know which exact route is being rendered

# Example

If pages directory is:
```
/pages/[group1]/index.astro
/pages/[group2]/index.astro
```

The rendered URLs would be `/foo/`, `/bar/`, etc.

There is currently no way to know if a URL belongs to either group `/[group1]/index.astro` or `/[group2]/index.astro`.

The current `Astro.url` is not enough to discern which group a URL belongs.

The proposed `Astro.route` would return `/[group1]/index.astro` or `/[group2]/index.astro` as fit.

**Comment by delucis** [comment]
+1 — this is something I’ve wanted in the past too.

**Comment by ematipico** [comment]
When you say "route" what exactly do you mean? The actual file path that matched the URL?

**Reply by aldo-roman** [reply]
Yes. Similar to how SvelteKit does it. See https://kit.svelte.dev/docs/load#using-url-data-route

So, if the file being rendered is `/src/pages/a/[b]/c/[d].astro` then `Astro.route` should return `/a/[b]/c/[d]` (without replacing `b` and `d`).

**Reply by delucis** [reply]
Yeah, basically the current route **pattern**.

**Comment by aldo-roman** [comment]
For anyone wondering, it was implemented in

https://github.com/withastro/astro/pull/11698
